src/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(QUICK_CUDA_CUDA_SOURCE gpu.cu gpu_type.cu gpu_get2e.cu)
+     
+set(QUICK_CUDA_FORTRAN_SOURCE gpu_write_info.f90)
+
+set_property(SOURCE ${QUICK_CUDA_FORTRAN_SOURCE} PROPERTY COMPILE_OPTIONS ${OPT_FFLAGS})
+
+#sadly FindCUDA doesn't support OBJECT libraries, so we have to use a static cuda_add_library
+cuda_add_library(quick_cuda STATIC ${QUICK_CUDA_FORTRAN_SOURCE} ${QUICK_CUDA_CUDA_SOURCE} 
+	OPTIONS ${CUDA_DEVICE_CODE_FLAGS} -DCUDA) 
+target_compile_options(quick_cuda PRIVATE -DCUDA)
```

Comment:
Why do we need `-DCUDA` twice?

Reply:
Oh yeah I can get rid of that.  It was because that library contained CUDA code (which only cares about the OPTIONS to cuda_add_library()), and Fortran code (which is affected by target_compile_options()).  But now it's a CUDA only library.